src/spikeinterface/preprocessing/deepinterpolation/train.py

```diff
@@ -0,0 +1,307 @@
+    verbose : bool
+        Whether to print the progress of the training, default is True
+    steps_per_epoch : int
+        Number of steps per epoch
+    period_save : int
+        Period of saving the model
+    apply_learning_decay : int
+        Whether to use a learning scheduler during training
+    nb_times_through_data : int
+        Number of times the data is repeated during training
+    learning_rate : float
+        Learning rate
+    loss : str
+        Loss function to be used
+    nb_workers : int
+        Number of workers to be used for the training
+    caching_validation : bool
+        Whether to cache the validation data, default is False
+    run_uid : str
+        Unique identifier for the training
+    network_name : str
+        Name of the network to be used, default is None
+    use_gpu : bool
+        Whether to use GPU, default is True
+    disable_tf_logger : bool
+        Whether to disable the tensorflow logger, default is True
+    memory_gpu : int
+        Amount of memory to be used by the GPU, default is None
+
+    Returns
+    -------
+    model_path : Path
+        Path to the model
+    """
+
+    if nb_workers == -1:
+        nb_workers = os.cpu_count()
+
+    args = (
+        recordings,
+        model_folder,
+        model_name,
+        desired_shape,
+        train_start_s,
+        train_end_s,
+        train_duration_s,
+        test_start_s,
+        test_end_s,
+        test_duration_s,
+        test_recordings,
+        pre_frame,
+        post_frame,
+        pre_post_omission,
+        existing_model_path,
+        verbose,
+        nb_gpus,
+        steps_per_epoch,
+        period_save,
+        apply_learning_decay,
+        nb_times_through_data,
+        learning_rate,
+        loss,
+        nb_workers,
+        caching_validation,
+        run_uid,
+        network_name,
+        use_gpu,
+        disable_tf_logger,
+        memory_gpu,
+    )
+    global train_func
+    train_func = train_deepinterpolation_process
+    with ProcessPoolExecutor(mp_context=mp.get_context("spawn"), max_workers=nb_workers) as executor:
+        f = executor.submit(train_func, *args)
+        model_path = f.result()
+    return model_path
+
+
+def train_deepinterpolation_process(
+    recordings: BaseRecording | list[BaseRecording],
+    model_folder: str | Path,
+    model_name: str,
+    desired_shape: tuple[int, int],
+    train_start_s: float,
+    train_end_s: float,
+    train_duration_s: float | None,
+    test_start_s: float,
+    test_end_s: float,
+    test_duration_s: float | None,
+    test_recordings: Optional[BaseRecording | list[BaseRecording]] = None,
+    pre_frame: int = 30,
+    post_frame: int = 30,
+    pre_post_omission: int = 1,
+    existing_model_path: Optional[str | Path] = None,
+    verbose: bool = True,
+    nb_gpus: int = 1,
+    steps_per_epoch: int = 10,
+    period_save: int = 100,
+    apply_learning_decay: int = 0,
```

Comment:
I would consider only exposing the parameters you care about, not all of them.

Reply:
For example, which parameters will in practice never change?

Reply:
you tell me :) In my opinion there's no harm in exposing everything with defaults!